javagen/src/main/java/com/azure/autorest/template/example/ClientMethodExampleWriter.java

```diff
@@ -64,6 +78,130 @@ public ClientMethodExampleWriter(ClientMethod method, String clientVarName, Prox
                         parameterInvocations));
 
         methodBodyWriter = javaBlock -> javaBlock.line(methodInvocation.toString());
+        responseAssertionWriter = methodBlock -> {
+            Optional<ProxyMethodExample.Response> responseOpt = proxyMethodExample.getPrimaryResponse();
+            if (responseOpt.isPresent()) {
+                ProxyMethodExample.Response response = responseOpt.get();
+                IType returnType = method.getReturnValue().getType();
+                if (returnType instanceof GenericType) {
+                    GenericType responseType = (GenericType) returnType;
+                    if (SyncPoller.class.getSimpleName().equals(responseType.getName())) {
+                        // SyncPoller<>
+
+                        if (response.getStatusCode() / 100 == 2) {
+                            // it should have a 202 leading to SUCCESSFULLY_COMPLETED
+                            // but x-ms-examples usually does not include the final result
+                            methodBlock.line("Assertions.assertEquals(LongRunningOperationStatus.SUCCESSFULLY_COMPLETED, response.waitForCompletion().getStatus());");
+                        }
+                    } else if (PagedIterable.class.getSimpleName().equals(responseType.getName())) {
+                        // PagedIterable<>
+
+                        // assert status code
+                        methodBlock.line(String.format("Assertions.assertEquals(%1$s, response.iterableByPage().iterator().next().getStatusCode());", response.getStatusCode()));
+                        // assert headers
+                        response.getHttpHeaders().stream().forEach(header -> {
+                            String expectedValueStr = ClassType.String.defaultValueExpression(header.getValue());
+                            String keyStr = ClassType.String.defaultValueExpression(header.getName());
+                            methodBlock.line(String.format("Assertions.assertEquals(%1$s, response.iterableByPage().iterator().next().getHeaders().get(HttpHeaderName.fromString(%2$s)).getValue());", expectedValueStr, keyStr));
+                        });
+                        // assert JSON of first item, or assert count=0
+                        if (ContentType.APPLICATION_JSON.equals(method.getProxyMethod().getRequestContentType())
```

Comment:
I may have made an error here. I think RequestContentType is actually content-type of request. Content-type of response should be `getResponseContentTypes`, and check if it contains "application/json"?

Reply:
Sure!